doc/source/ray-air/examples/vicuna_13b_lightning_deepspeed_finetune.ipynb

```diff
@@ -0,0 +1,1468 @@
+     ]
+    }
+   ],
+   "source": [
+    "for case in testcases:\n",
+    "    prompt = PROMPT_TEMPLATE.format(intent=case[\"intent\"], snippet=\"\")\n",
+    "    output = generator(prompt, max_new_tokens=30, do_sample=True)\n",
+    "    print(output[0][\"generated_text\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Test the Generated Code Snippets\n",
+    "\n",
+    "The generated code snippets look pretty reasonable. The results covered Pandas operations, regular expressions, and Linux commands. Let's test them one by one."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Before\n",
+      "            col\n",
+      "0  abc def ghi\n",
+      "1     12 3 456\n",
+      "2             \n",
+      "After\n",
+      "            col\n",
+      "0  abc_def_ghi\n",
+      "1    _12_3_456\n",
+      "2        _____\n"
+     ]
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "df = pd.DataFrame.from_dict({\"col\": [\"abc def ghi\", \" 12 3 456\", \"     \"]})\n",
+    "print(\"Before\\n\", df)\n",
+    "\n",
+    "df[\"col\"] = df[\"col\"].str.replace(\" \", \"_\")\n",
+    "print(\"After\\n\", df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['>The Great Gatsby<',\n",
+       " '>F. Scott Fitzgerald<',\n",
+       " '>1925<',\n",
+       " '>Sapiens: A Brief History of Humankind<',\n",
+       " '>Yuval Noah Harari<',\n",
+       " '>2011<']"
+      ]
+     },
+     "execution_count": 47,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import re\n",
+    "\n",
+    "line = \"\"\"\n",
+    "<bookstore>\n",
+    "  <book category=\"fiction\">\n",
+    "    <title>The Great Gatsby</title>\n",
+    "    <author>F. Scott Fitzgerald</author>\n",
+    "    <year>1925</year>\n",
+    "  </book>\n",
+    "  <book category=\"non-fiction\">\n",
+    "    <title>Sapiens: A Brief History of Humankind</title>\n",
+    "    <author>Yuval Noah Harari</author>\n",
+    "    <year>2011</year>\n",
+    "  </book>\n",
+    "</bookstore>\n",
+    "\"\"\"\n",
+    "re.findall(\">.*<\", line)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, let's hand it over to LLM and let it wrap up the demo:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2023-06-30 19:18:30,407\tWARNING tune.py:192 -- Stop signal received (e.g. via SIGINT/Ctrl+C), ending Ray Tune run. This will try to checkpoint the experiment state one last time. Press CTRL+C (or send SIGINT/SIGKILL/SIGTERM) to skip. \n"
+     ]
```

Comment:
Why is Tune running here?

Reply:
It seems to be another information leak of Tune... I'll create a separate issue for this